x/gateway/keeper/msg_server_stake_gateway_test.go

```diff
@@ -60,33 +61,61 @@ func TestMsgServer_StakeGateway_FailLoweringStake(t *testing.T) {
 	k, ctx := keepertest.GatewayKeeper(t)
 	srv := keeper.NewMsgServerImpl(k)
 
-	// Prepare the gateway
+	// Prepare the gateway.
 	addr := sample.AccAddress()
-	initialStake := sdk.NewCoin("upokt", math.NewInt(100))
-	stakeMsg := &types.MsgStakeGateway{
+	initialStake := cosmostypes.NewCoin("upokt", math.NewInt(100))
+	stakeMsg := &gatewaytypes.MsgStakeGateway{
 		Address: addr,
 		Stake:   &initialStake,
 	}
 
-	// Stake the gateway & verify that the gateway exists
+	// Stake the gateway & verify that the gateway exists.
 	_, err := srv.StakeGateway(ctx, stakeMsg)
 	require.NoError(t, err)
 	_, isGatewayFound := k.GetGateway(ctx, addr)
 	require.True(t, isGatewayFound)
 
-	// Prepare an updated gateway with a lower stake
-	updatedStake := sdk.NewCoin("upokt", math.NewInt(50))
-	updateMsg := &types.MsgStakeGateway{
+	// Prepare an updated gateway with a lower stake.
+	updatedStake := cosmostypes.NewCoin("upokt", math.NewInt(50))
+	updateMsg := &gatewaytypes.MsgStakeGateway{
 		Address: addr,
 		Stake:   &updatedStake,
 	}
 
-	// Verify that it fails
+	// Verify that it fails.
 	_, err = srv.StakeGateway(ctx, updateMsg)
 	require.Error(t, err)
 
-	// Verify that the gateway stake is unchanged
+	// Verify that the gateway stake is unchanged.
 	gatewayFound, isGatewayFound := k.GetGateway(ctx, addr)
 	require.True(t, isGatewayFound)
 	require.Equal(t, initialStake.Amount, gatewayFound.Stake.Amount)
 }
+
+func TestMsgServer_StakeGateway_FailBelowMinStake(t *testing.T) {
+	k, ctx := keepertest.GatewayKeeper(t)
+	srv := keeper.NewMsgServerImpl(k)
+
+	addr := sample.AccAddress()
+	gatewayStake := cosmostypes.NewInt64Coin(volatile.DenomuPOKT, 100)
+	minStake := gatewayStake.AddAmount(math.NewInt(1))
+	expectedErr := gatewaytypes.ErrGatewayInvalidStake.Wrapf("gateway %q must stake at least %s", addr, minStake)
+
+	// Set the minimum stake to be greater than the gateway stake.
+	params := k.GetParams(ctx)
+	params.MinStake = &minStake
+	err := k.SetParams(ctx, params)
+	require.NoError(t, err)
+
+	// Prepare the gateway.
+	stakeMsg := &gatewaytypes.MsgStakeGateway{
+		Address: addr,
+		Stake:   &gatewayStake,
+	}
+
+	// Attempt to stake the gateway & verify that the gateway does NOT exist.
+	_, err = srv.StakeGateway(ctx, stakeMsg)
+	require.ErrorContains(t, err, expectedErr.Error())
```

Comment:
Personally, I think that checking if `ErrorContains(atewaytypes.ErrGatewayInvalidStake)` is enough so you don't have to update the test if the error string message changes.

Reply:
My personal preference is that the test be sensitive to the error message changing as this could be indicative of a couple things:

1. Business logic changed such that a different error is returned for a given scenario. In many cases, this could be the same error type (e.g. `ErrAppParamInvalid`) which is wrapping (`.Wrapf(...)`) a message from a different error path. 

    This could lead to false negatives (i.e. tests which pass but should not) appearing over time as the codebase evolves.
2. The error or wrapped message was changed. 

    In this case, I think we should want to have coverage over the error message such that we're confident that our expectation of how it looks is accurate. We're likely going to encounter these logs and error messages in bug reports, which need to be immediately useful.